study1/week6/LengthOfLongestSubstring.java

```diff
@@ -0,0 +1,36 @@
+import java.lang.Math;
+
+class LengthOfLongestSubstring {
+    public int lengthOfLongestSubstring(String s) {
+
+        Map<Character, Integer> charList = new HashMap<>();
+        int res = 0;
+        int tmpSize = 0;
+
+        for (int cur = 0; cur < s.length(); cur++) {
+
+            if (charList.containsKey(s.charAt(cur))) {
+                res = Math.max(tmpSize, res);
+                tmpSize = 0;
+                cur = charList.get(s.charAt(cur));
+                charList.clear();
```

Comment:
Clearing a datstructure within an algorithm is quite inefficient. Rather, you can try to track where the substring starts.